src/rules/ResourceCompleteness.js

```diff
@@ -17,37 +17,31 @@
  * limitations under the License.
  */
 
-import { Rule, Result } from "i18nlint-common";
+import { Result } from "i18nlint-common";
 import Locale from "ilib-locale";
 
+import ResourceRule from './ResourceRule.js';
+
 /** Rule to check that a resource has both source and target elements */
-class ResourceCompleteness extends Rule {
+class ResourceCompleteness extends ResourceRule {
     /** @readonly */ name = "resource-completeness";
     /** @readonly */ description = "Ensure that resources are complete, i.e. have both source and target elements.";
     /** @readonly */ link = "https://github.com/ilib-js/i18nlint/blob/main/docs/resource-completeness.md";
-    /** @readonly */ ruleType = "resource";
-
-    constructor() {
-        super({});
-    }
 
-    /** @override */
-    getRuleType() {
-        return this.ruleType;
+    constructor(options) {
+        super(options);
     }
 
-    /** Check that a given resource has both source and target tags set
+    /**
+     * Check that a given resource has both source and target tags set
      * @override
-     * @param {object} options
-     * @param {import("ilib-tools-common").Resource} options.resource
-     * @param {string} options.file
-     * @param {string} options.locale
-     *
+     * @param {string} source the source string
```

Comment:
Please restore `@param {object} options` (or `params` if You prefer) and the notation `options.` to fix type annotations